cli/view.js

```diff
@@ -75,28 +75,39 @@ const loadAndAddHandlers = ({app, handlersArg, datasetDir, narrativeDir}) => {
     `Looking for datasets in ${datasetsPath}\nLooking for narratives in ${narrativesPath}`;
 };
 
-const getAuspiceBuild = () => {
+const getAuspiceBuild = (customBuildOnly) => {
   const cwd = path.resolve(process.cwd());
   const sourceDir = path.resolve(__dirname, "..");
-  if (
-    cwd !== sourceDir &&
-    fs.existsSync(path.join(cwd, "index.html")) &&
-    fs.existsSync(path.join(cwd, "dist")) &&
-    fs.readdirSync(path.join(cwd, "dist")).filter((fn) => fn.match(/^auspice.bundle.[a-z0-9]+.js$/)).length === 1
-  ) {
-    return {
-      message: "Serving the auspice build which exists in this directory.",
-      baseDir: cwd,
-      distDir: path.join(cwd, "dist")
-    };
+
+  // Default to current working directory.
+  let baseDir = cwd;
+  if (!hasAuspiceBuild(cwd)) {
+    if (cwd === sourceDir || customBuildOnly) {
+      utils.error(`Auspice build files not found under ${cwd}. Did you run \`auspice build\` in this directory?`);
+      process.exit(1);
+    } else if (!hasAuspiceBuild(sourceDir)) {
+      utils.error(`Auspice build files not found under ${cwd} or ${sourceDir}. Did you run \`auspice build\` in either directory?`);
+      process.exit(1);
+    } else if (cwd !== sourceDir) {
+      utils.log(`Auspice build files not found under ${cwd}. Using build files under ${sourceDir}.`)
+      baseDir = sourceDir;
+    }
   }
   return {
-    message: `Serving auspice version ${version}`,
-    baseDir: sourceDir,
-    distDir: path.join(sourceDir, "dist")
+    message: `Serving the auspice build in ${baseDir}.`,
+    baseDir: baseDir,
+    distDir: path.join(baseDir, "dist")
   };
 };
 
+const hasAuspiceBuild = (directory) => {
+  return (
+    fs.existsSync(path.join(directory, "dist")) &&
+    fs.existsSync(path.join(directory, "dist/index.html")) &&
+    fs.readdirSync(path.join(directory, "dist")).filter((fn) => fn.match(/^auspice.bundle.[a-z0-9]+.js$/)).length === 1
```

Comment:
As noted by @tsibley  in #1520:

> those . should be escaped in the regex since they're meant to be literals.

Reply:
Fixed in 4b9ca4dc416daca9b231162d70bead3b01259c42.